ext/standard/crc32.c

```diff
@@ -23,19 +23,30 @@
 # if defined(__linux__)
 #  include <sys/auxv.h>
 #  include <asm/hwcap.h>
+# elif defined(__FreeBSD__)
+#  include <sys/auxv.h>
+#  include <elf.h>
 # endif
 
 static inline int has_crc32_insn() {
 	/* Only go through the runtime detection once. */
 	static int res = -1;
 	if (res != -1)
 		return res;
+# if defined(__linux__)
 # if defined(HWCAP_CRC32)
 	res = getauxval(AT_HWCAP) & HWCAP_CRC32;
 	return res;
 # elif defined(HWCAP2_CRC32)
 	res = getauxval(AT_HWCAP2) & HWCAP2_CRC32;
 	return res;
+# endif
+# elif defined(__FreeBSD__)
+# if defined(HWCAP_CRC32)
+	zend_long crc32 = 0;
+	if (elf_aux_info(AT_HWCAP, &crc32, sizeof(crc32)) == 0)
+		res = crc32 & HWCAP_CRC32;
```

Comment:
`res` isn't returned here. This will only work on the second run.

Reply:
ah I forgot about this one but a lot had changed since and it s an useless change now, FreeBSD uses a wrapper to be closer to Linux.